src/gui/editors/PianoRoll.cpp

```diff
@@ -2609,6 +2609,7 @@ void PianoRoll::dragNotes( int x, int y, bool alt, bool shift, bool ctrl )
 			{
 				if (note->selected())
 				{
+					if (note->oldLength() <= 0){note->setOldLength(1);}
```

Comment:
```suggestion
					if (note->oldLength() <= 0){note->setOldLength(4);}
```
Beat notes appear to be drawn four ticks long, so `setOldLength(1)` leads to them shrinking at first when scaled.